src/python/pants/backend/python/goals/coverage_py.py

```diff
@@ -506,14 +506,16 @@ async def merge_coverage_data(
 
 @rule(desc="Generate Pytest coverage reports", level=LogLevel.DEBUG)
 async def generate_coverage_reports(
-    merged_coverage_data: MergedCoverageData,
+    data_collection: PytestCoverageDataCollection,
```

Comment:
So we have a rule that matches the polymorphic signature (takes a subclass of CoverageDataCollection and returns a CoverageReports).